lib/vector-core/src/stream/driver.rs

```diff
@@ -220,10 +219,18 @@ where
                             bytes_sent.emit(ByteSize(byte_size));
                         }
                     }
-                    events_sent.emit(response.events_sent());
+
+                    if let Some(sizes) = response.events_sent().sizes() {
+                        for (tags, size) in sizes {
+                            events_sent.emit(tags, *size);
+                        }
+                    } else if let Some(size) = response.events_sent().size() {
+                        events_sent.emit(&(OptionalTag::Ignored, OptionalTag::Ignored), size);
+                    }
```

Comment:
Would this make sense to move into a function on `CachedEvent`, leaving this call site unchanged?

Reply:
Agreed: the logic is a bit fuzzy to stick in here. It made me do a double take when I first read it because of how duplicative it is.

Pushing this back up to CachedEvent/TaggedEventsSent would be nice, because then it could also potentially just get turned into a regular old match statement and avoid the question of "well what if both of those conditions return None?"

Reply:
It doesn't make sense to add it to `CachedEvent` since the complication is in `RequestCountByteSize` being able to hold a single or multiple counts. I'd have to create some kind of `SingleOrMultiple` trait for `CachedEvent` to use. I think that would be an over complication.

I've moved it to a function on `RequestCountByteSize` instead.